src/commands/prompt/run.ts

```diff
@@ -1,93 +1,46 @@
+import type { DisplayMessage } from '../../interface/chat/chat-message.js';
+import { renderChatInterface } from '../../interface/chat/index.js';
 import * as output from '../../output.js';
-import { run as init } from '../init/index.js';
-import { processCommand } from './commands.js';
-import type { PromptOptions, SessionContext } from './types.js';
-import {
-  filterOutApiKey,
-  getDefaultProvider,
-  getOutputParams,
-  handleInputFile,
-  resolveProviderFromOption,
-} from './utils.js';
+import type { PromptOptions, SessionContext, SessionFeedback } from './types.js';
+import { getDefaultProvider, handleInputFile, resolveProviderFromOption } from './utils.js';
 
 export async function run(initialPrompt: string, options: PromptOptions) {
   try {
-    await runInternal(initialPrompt.trim(), options);
+    const session = await createSession(options, initialPrompt);
+    renderChatInterface(session);
   } catch (error) {
     output.clearLine();
     output.outputError(error);
     process.exit(1);
   }
 }
 
-async function runInternal(initialPrompt: string, options: PromptOptions) {
-  output.setVerbose(options.verbose);
-
-  const configExists = await checkIfConfigExists();
-  if (!configExists) {
-    await init();
-    return;
-  }
-
-  const session = await createSession(options);
-
-  if (options.file) {
-    handleInputFile(session, options.file);
-  }
-
-  if (initialPrompt) {
-    output.outputUser(initialPrompt);
-    await handleMessage(session, initialPrompt);
-  } else {
-    output.outputAi('Hello, how can I help you?');
-  }
-
-  if (options.interactive || !initialPrompt) {
-    output.outputInfo(
-      'Type "/exit" or press Ctrl+C to exit. Type "/help" to see available commands.',
-    );
-  } else {
-    process.exit(0);
-  }
-
-  // eslint-disable-next-line no-constant-condition
-  while (true) {
-    const userPrompt = await inputLine('me: ');
-    const isCommand = processCommand(session, userPrompt);
-    if (isCommand) {
-      continue;
-    }
-
-    await handleMessage(session, userPrompt);
-  }
-}
-
-async function createSession(options: PromptOptions): Promise<SessionContext> {
+async function createSession(
+  options: PromptOptions,
+  initialPrompt?: string,
+): Promise<SessionContext> {
   const configFile = await parseConfigFile();
-  output.outputVerbose(`Config: ${JSON.stringify(configFile, filterOutApiKey, 2)}`);
-  output.setShowStats(options.stats ?? configFile.showStats);
-  output.setShowCosts(options.costs ?? configFile.showCosts);
 
   const provider = options.provider
     ? resolveProviderFromOption(options.provider)
     : getDefaultProvider(configFile);
-  output.outputVerbose(`Using provider: ${provider.label}`);
 
   const initialConfig = configFile.providers[provider.name];
   if (!initialConfig) {
     throw new Error(`Provider config not found: ${provider.name}.`);
   }
 
   let style = {};
+  const sessionFeedback: SessionFeedback = {};
```

Comment:
I would have a unfiied list of output consisting of both Messages and Output from AI. That would match the UI paradigm we have and should reduce the complexity in the UI components

Reply:
Done